### ISSUE TITLE
Compare hashes to avoid re-downloading.

### ISSUE THREAD

**Issue by ansdma** [pyenv/pyenv#28]
This is just a suggestion. 
Why not before downloading, check if the requested version has already been downloaded and compare hashes. This can save some time when build fails due to missing libs.

**Comment by opscolin** [comment]
sorry about using pyevn install python after I downloaded the source package 

my host is "CentOS Linux release 7.1.1503 (Core) "

for installation is too slow, so I downloaded the source package(Python-3.4.4.tgz via url https://www.python.org/ftp/python/3.4.4/Python-3.4.4.tgz) 
and moved it to /root/.pyenv/cache/ folder(I created by hand) and then re-try to use 'pyenv install 3.4.4' to install, but it still downloads the source package and slow too.

so what's the problem? where did I do wrong ? or something else ?

Thanks

**Comment by opscolin** [comment]
sorry~

after detail check and find the root problem

when using `pyenv install 3.4.4 -v`, it showed : 
```
Downloading Python-3.4.4.tgz...
-> https://www.python.org/ftp/python/3.4.4/Python-3.4.4.tgz
```

so I downloaded file `Python-3.4.4.tgz` and moved it to `~/.pyenv/cache/` folder. it didn't work.

but when I checked that there was a file named `Python-3.4.4.tar.gz` under /tmp/python.xxxxxxx.xxxx/

so I changed name from `Python-3.4.4.tgz` to `Python-3.4.4.tar.gz` under `~/.pyenv/cache/` folder, and re-try to install, it worked .

Thanks too.
